core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -42,6 +42,18 @@ private[ui] class AllJobsPage(parent: JobsTab) extends WebUIPage("") {
     }
 
     def makeRow(job: JobUIData): Seq[Node] = {
+      // scalastyle:off
+      val killLink = if (killEnabled) {
+        val killLinkUri = s"${UIUtils.prependBaseUri(parent.basePath)}/jobs/job/kill?" +
+          s"id=${job.jobId}&terminate=true"
+        val confirm = "return window.confirm(" +
+          s"'Are you sure you want to kill job ${job.jobId} ?');"
+        <span class="kill-link">
+          (<a href={killLinkUri} onclick={confirm}>kill</a>)
+        </span>
+      }
+      // scalastyle:on
```

Comment:
yes, if killEnabled = false, the type of killLink is Any.
i will add an else case that returns Seq.empty.